src/extension.ts

```diff
@@ -21,6 +23,41 @@ import { LeanTaskGutter, LeanTaskMessages } from './taskgutter';
 import { StaticServer } from './staticserver';
 import { LibraryNoteLinkProvider } from './librarynote';
 
+function executablePath(): string {
+    return workspace.getConfiguration('lean').get('lean.executablePath', 'lean')
+}
+
+async function checkLean3(): Promise<boolean> {
+    const folders = workspace.workspaceFolders
```

Comment:
There is a lot of code in `src/server.ts` that detects whether we should offer to install elan, or whether we need to do some workarounds on windows.  This should probably be done here instead or shared with the server code.

Reply:
I have a different proposal. What if `checkLean3` becomes the following:
```typescript
async function checkLean3(): Promise<boolean> {
    const lean4 = extensions.getExtension('jroesch.lean4');
    if (!lean4) {
        return true;
    }
    return (await lean4.activate()).aborted;
}
```
vscode-lean4 would provide an API for `aborted`, which is the result of a similar version check, but for Lean 4. I already tried this out and the effects are as follows:
- If vscode-lean4 is not installed, vscode-lean behaves exactly as before.
- If both are installed while the corresponding local toolchain is not and a Lean 3 project is opened, vscode-lean takes a little longer to activate while the elan invocation of vscode-lean4 downloads the corresponding Lean version. If it does not succeed at this for whatever reason, vscode-lean4 emits an error where it is visibly obvious that it was produced by vscode-lean4 and the behaviour of vscode-lean remains exactly as before.
- The changes to vscode-lean are minimal.
- My concern about concurrent invocations of elan from above is resolved.

Thoughts? If the longer activation time without visual cues is an issue, I could try to find a way to stream the elan stderr to the vscode output in vscode-lean4, similar to what vscode-lean does to show that work is being done.

Reply:
This approach sounds great to me.  Most of the corner cases with the Lean 3 extensions are for newcomers, which hopefully don't use Lean 4 yet.

One thing: please don't use `jroesch.lean4` for the new extension.  Instead use `leanprover.lean4`.  If you tell me the email address you use for your azure account, then I can make you a co-owner of the leanprover publisher id.